src/PFire.Core/TcpServer.cs

```diff
@@ -23,13 +23,15 @@ public class TcpServer
 
         private readonly TcpListener _listener;
         private bool _running;
+        private PFireServer _server;
 
-        public TcpServer(IPEndPoint endPoint)
+        public TcpServer(IPEndPoint endPoint, PFireServer server)
```

Comment:
The idea to avoid the `PFireServer` here is to avoid coupling our domaint to the infrastructure of the communication protocol